tests/test_vqe.py

```diff
@@ -707,6 +707,23 @@ def test_simplify(self, old_H, new_H):
         old_H.simplify()
         assert old_H.compare(new_H)
 
+    def test_simplify_while_queueing(self):
+        """Tests that simplifying a Hamiltonian in a tape context
+        queues the simplified Hamiltonian."""
+
+        with qml.tape.QuantumTape() as tape:
+            a = qml.PauliX(wires=0)
+            b = qml.PauliY(wires=1)
+            c = qml.Identity(wires=2)
+            d = b @ c
+            H = qml.Hamiltonian([1., 2.], [a, d])
+            H.simplify()
+
+        # check that H is simplified
+        assert H.ops == [a, b]
+        # check that the simplified Hamiltonian is in the queue
+        assert H in tape._queue
```

Comment:
Since H is not given to a measurement, the owners of H are not updated... in other words, the queue is pretty messed up in this example, but without the changes to this PR this tape would throw an error in the first place!